### PR TITLE
#WIP #CU-2nhgxne BYOG pallet version with custom header

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Description
* if user will provide header and we allow pay for tx in assets -> user will pay
* Bring your own gas - pay of tx in foreign assets. Reading this you will be able to grasp current state of feature and be able progress implementing (of course if you are rust substrate developer) and as UX person get what to expect and tell users.
* https://github.com/paritytech/substrate/pull/12087
* https://github.com/paritytech/cumulus/pull/1572
* https://github.com/ComposableFi/substrate/pull/12
* https://github.com/ComposableFi/cumulus/pull/6

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2nhgxne XCM transfers - gas tokens of destination parachain [pallet]](https://app.clickup.com/t/2nhgxne)

File: docs/byog.md [hunk 1]
```diff
@@ -6,17 +6,17 @@ BYOG allows owners of nonnative tokens to execute transactions by paying fees in
...
-`On ramps` and [meta transactions](https://docs.polygon.technology/docs/category/meta-transactions) handle such possibilities on other chains. 
+`On ramps` and [meta (gas less) transactions](https://docs.polygon.technology/docs/category/meta-transactions) handle such possibilities on other chains.
 
 ## Current state
 
-Any cross-chain (XCMP) transactions will choose what assets to use to pay for the transaction.
+Any cross-chain (XCMP) transactions can choose what assets to use to pay for the transaction.
 
-Direct native transactions will check several hardcoded foreign assets by a predefined order and try to pay using these on behalf of a user.
+Direct transaction can add header to pick up `payment` asset.
```
Comment (PoisonPhang) [inline docs/byog.md:15]: NIT
```suggestion
Direct transactions can add a header to pick up some `payment` asset. 
```

**Review by PoisonPhang — APPROVED** [review]
LGTM!
One NIT comment

**Review by seunlanlege — APPROVED** [review]

**Review by PoisonPhang — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
Likely tests fails as we use 0.9.22 cumulus and we use statemine from that to run multu assets tests xcmp, so need update too https://github.com/ComposableFi/cumulus

**Comment by dzmitry-lahoda** [comment]
@Mergifyio queue

**Pull request closed without merging**